core/asset.go

```diff
@@ -1,9 +1,11 @@
 package core
 
-import "regexp"
+import (
+	"regexp"
+)
 
-func AssetIsValid(v string) bool {
-	re := regexp.MustCompile("[A-Z]{1,8}")
-
-	return re.Match([]byte(v))
+// AssetIsValid
+func AssetIsValid(asset string) bool {
+	valid, _ := regexp.MatchString("^[A-Z]{1,16}$", asset)
```

Comment:
Panic if the regexp is not valid seems a good point here.
Even initialize the regexp as global package var would be a good idea. The binary will panic immediately if not valid.
Agree with the fact than today, it works properly, but maybe, the charset could be extended.

Reply:
The pull request got updated following your recommendation. Feels free to let me know if that way is something you expected to be implemented :)